library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -693,79 +708,78 @@ def ConcealSecrets(devs)
     # @return true if sucess
     def Read
       return true if @initialized == true
 
       @Devices = {}
 
       # preparation
-      allfiles = SCR.Dir(path(".network.section"))
-      allfiles = [] if allfiles.nil?
-      devices = Builtins.filter(allfiles) do |file|
-        !Builtins.regexpmatch(file, "[~]")
-      end
-      Builtins.y2debug("devices=%1", devices)
+      devices = get_devices
 
       # Read devices
-      Builtins.maplist(devices) do |d|
-        pth = Ops.add(Ops.add(".network.value.\"", d), "\"")
-        Builtins.y2debug("pth=%1", pth)
-        values = SCR.Dir(Builtins.topath(pth))
-        Builtins.y2debug("values=%1", values)
-        config = {}
-        Builtins.maplist(values) do |val|
-          item = Convert.to_string(
-            SCR.Read(Builtins.topath(Ops.add(Ops.add(pth, "."), val)))
-          )
-          Builtins.y2debug("item=%1", item)
-          next if item.nil?
-          # No underscore '_' -> global
-          # Also temporarily standard globals
-          if Ops.less_than(Builtins.find(val, "_"), 0) ||
-              LOCALS.include?(val)
-            Ops.set(config, val, item)
-            next
-          end
-          # Try to strip _suffix
-          v = Builtins.substring(val, 0, Builtins.findlastof(val, "_"))
-          s = Builtins.substring(val, Builtins.findlastof(val, "_"))
-          s = Builtins.substring(s, 1) if Ops.greater_than(Builtins.size(s), 1)
-          Builtins.y2milestone("%1:%2:%3", val, v, s)
-          # Global
-          if !LOCALS.include?(v)
-            Ops.set(config, val, item)
-          else
-            aliases = Ops.get_map(config, "_aliases", {})
-            suf = Ops.get_map(aliases, s, {})
-            Ops.set(suf, v, item)
-            Ops.set(aliases, s, suf)
-            Ops.set(config, "_aliases", aliases)
-          end
-        end
-        Builtins.y2milestone("config=%1", ConcealSecrets1(config))
-        # canonicalize, #46885
-        caliases = Builtins.mapmap(Ops.get_map(config, "_aliases", {})) do |a, c|
-          { a => CanonicalizeIP(c) }
-        end
-        config["_aliases"] = caliases if caliases != {}
-        config = CanonicalizeIP(config)
-        config = CanonicalizeStartmode(config)
-        config = filter_interfacetype(config)
+      devices.each do |d|
+        pth = ".network.value.\"#{d}\""
+        log.debug("pth=#{pth}")
+
+        values = SCR.Dir(path(pth))
+        log.debug("values=#{values}")
 
-        devtype = GetTypeFromIfcfg(config)
-        devtype = GetType(d) if devtype.nil?
+        config = generate_config(pth, values)
 
-        dev = @Devices[devtype] || {}
-        dev[d] = config
-        @Devices[devtype] = dev
+        set_devtype_config(d, config)
       end
-      Builtins.y2debug("Devices=%1", @Devices)
+      log.debug("Devices=#{@Devices}")
 
       @OriginalDevices = deep_copy(@Devices)
       @initialized = true
-      true
+    end
+
+    def Reset
```

Comment:
I'm not that sure that creating method which is used only in test is good idea. Or did I overlooked something again?